File: Leetcode/28.Implement_strStr.cpp
class Solution {
public:
    int strStr(string haystack, string needle) {
        if (needle.empty())
            return 0;
        else if (haystack.empty())
            return -1;
        
        if (haystack.size() < needle.size())
            return -1;
            
        for(int i=0;i<haystack.size()-needle.size()+1;i++)
        {
            int j;
            for(j=0;j<needle.size();j++)
            {
                if(haystack[i+j]!=needle[j])
                    break;
            }
            if(j==needle.size())
                return i;
        }
        return -1;
    }
};

File: Leetcode/28.Implement_strStr.py
class Solution(object):
    def strStr(self, haystack, needle):
        """
        :type haystack: str
        :type needle: str
        :rtype: int
        """
        #表示没有输入或输入错误的情况
        if haystack is None or needle is None:
            return -1
        for i in range(len(haystack)-len(needle)+1):
            for j in range(len(needle)):
                if haystack[i+j]!=needle[j]:
                    break
            else:
                return i
        return -1
